### PR TITLE
drm/atomic: Update connector prop mapping in `reset_state()`, and refactor prop mapping code a bit

### PR DESCRIPTION
**Pull request by ids1024**
Fixes https://github.com/Smithay/smithay/issues/1374.

### CODE REVIEW

**Comment by YaLTeR** [comment]
If I store some prop handles in my compositor (i.e. GAMMA_LUT) does that mean I need to refresh those at some point too, and when?

File: src/backend/drm/surface/atomic.rs [hunk 1]
```diff
@@ -90,18 +90,9 @@ impl State {
         // If they don't match, `commit_pending` will return true and they will be changed on the next `commit`.
         let mut current_connectors = HashSet::new();
         // make sure the mapping is up to date
-        map_props(fd, res_handles.connectors(), &mut prop_mapping.0)?;
+        map_props(fd, res_handles.connectors(), &mut prop_mapping.connectors)?;
```
Comment (Drakulix) [inline src/backend/drm/surface/atomic.rs:93]: If we want be consistent here, we should probably clear `prop_mapping.connectors` first to not keep old entries around of properties missing for a new connector, no?

**Comment by Drakulix** [comment]
> If I store some prop handles in my compositor (i.e. GAMMA_LUT) does that mean I need to refresh those at some point too, and when?

You should remove and re-evaluate them on hotplug, so if you are using smithay-drm-extras on `connector_changed` (otherwise in `UdevHandler::device_changed`, for any new connectors).

Note this only applies to connector properties, crtcs and planes are static anyhow.

**Comment by ids1024** [comment]
> If I store some prop handles in my compositor (i.e. GAMMA_LUT) does that mean I need to refresh those at some point too, and when?

This is specifically an issue with connectors, which on some cards can be added and removed. I think once a drm object is created the properties it has are static? And I don't know if planes or CRTCs are ever added dynamically...

**Comment by YaLTeR** [comment]
Okay I'm not sure I actually store any connector properties, but either way I would store them for each connector as it is connected (and remove that info when it is disconnected). But I do call `drm_compositor.reset_state()` which is mentioned in the title of this PR, which is why I asked.

**Review by Drakulix — APPROVED** [review]
LGTM now

**Pull request merged**